File: pkg/java/src/main/java/dev/openfga/language/errors/ErrorProperties.java
package dev.openfga.language.errors;

public class ErrorProperties {

    private StartEnd line;

    private StartEnd column;

    private String file;

    private String module;

    private String message;

    public ErrorProperties(StartEnd line, StartEnd column, String message) {
        this.line = line;
        this.column = column;
        this.message = message;
    }

    public ErrorProperties(StartEnd line, StartEnd column, String message, String file, String module) {
        this.line = line;
        this.column = column;
        this.message = message;
        this.file = file;
        this.module = module;
    }

    String getFullMessage(String type) {
        if (line != null && column != null) {
            return String.format(
                    "%s error at line=%d, column=%d: %s", type, line.getStart(), column.getStart(), message);
        } else {
            return String.format("%s error: %s", type, message);
        }
    }

    public StartEnd getLine() {
        return line;
    }

    public void setLine(StartEnd line) {
        this.line = line;
    }

    public StartEnd getColumn() {
        return column;
    }

    public void setColumn(StartEnd column) {
        this.column = column;
    }

    public String getFile() {
        return file;
    }

    public void setFile(String file) {
        this.file = file;
    }

    public String getModule() {
        return module;
    }

    public void setModule(String module) {
        this.module = module;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: pkg/csharp/src/OpenFga.Language/Errors/ErrorProperties.cs
namespace OpenFga.Language.Errors;

/// <summary>
/// Contains properties that describe an error including location information and error message.
/// Used as a base class for various error types to provide consistent error reporting.
/// </summary>
public class ErrorProperties {
    /// <summary>
    /// Gets or sets the line range where the error occurred.
    /// </summary>
    public StartEnd? Line { get; set; }

    /// <summary>
    /// Gets or sets the column range where the error occurred.
    /// </summary>
    public StartEnd? Column { get; set; }

    /// <summary>
    /// Gets or sets the error message describing what went wrong.
    /// </summary>
    public string Message { get; set; } = string.Empty;

    /// <summary>
    /// Initializes a new instance of the ErrorProperties class.
    /// </summary>
    public ErrorProperties() { }

    /// <summary>
    /// Initializes a new instance of the ErrorProperties class with location and message.
    /// </summary>
    /// <param name="line">The line range where the error occurred</param>
    /// <param name="column">The column range where the error occurred</param>
    /// <param name="message">The error message</param>
    public ErrorProperties(StartEnd? line, StartEnd? column, string message) {
        Line = line;
        Column = column;
        Message = message;
    }

    /// <summary>
    /// Gets a formatted error message including the error type and location information.
    /// </summary>
    /// <param name="type">The type of error (e.g., "syntax", "validation")</param>
    /// <returns>A formatted error message</returns>
    internal string GetFullMessage(string type) {
        if (Line != null && Column != null) {
            return $"{type} error at line={Line.Start}, column={Column.Start}: {Message}";
        }
        else {
            return $"{type} error: {Message}";
        }
    }
}